src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -689,13 +686,13 @@ internal static bool IsZeroElementTupleType(TypeSymbol type)
                 var subPattern = node.Subpatterns[i];
                 bool isError = outPlaceholders.IsDefaultOrEmpty || i >= outPlaceholders.Length;
                 TypeSymbol elementType = isError ? CreateErrorType() : outPlaceholders[i].Type;
-                ParameterSymbol parameter = null;
+                ParameterSymbol? parameter = null;
                 if (subPattern.NameColon != null && !isError)
                 {
                     // Check that the given name is the same as the corresponding parameter of the method.
-                    string name = subPattern.NameColon.Name.Identifier.ValueText;
+                    string? name = subPattern.NameColon.Name.Identifier.ValueText;
                     int parameterIndex = i + skippedExtensionParameters;
-                    if (parameterIndex < deconstructMethod.ParameterCount)
+                    if (parameterIndex < deconstructMethod!.ParameterCount)
```

Comment:
>deconstructMethod!. [](start = 41, length = 19)

I didn't follow how we know that `deconstructMethod` is not null here.

Reply:
What was the resolution here?

---
In reply to: [380984757](https://github.com/dotnet/roslyn/pull/41503#discussion_r380984757) [](ancestors = 380984757)

Reply:
It is a logical conclusion from the code above.  Line 688 sets hasErrors when deconstructMethod is null.  Line 697 prevents us getting to this line when hasErrors is true.  Therefore at this line we know it is not null.

---
In reply to: [384027925](https://github.com/dotnet/roslyn/pull/41503#discussion_r384027925) [](ancestors = 384027925,380984757)

Reply:
I'm sorry, but I'm still missing something. We have `if (deconstructMethod is null) hasErrors = true;`, but `hasErrors` doesn't seem to be used elsewhere in this method.